sdk/cosmosdb/cosmos/src/diagnostics/CosmosException.ts

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { ErrorResponse } from "..";
+import { getCosmosDiagnosticsToString, setDiagnostics } from "./Diagnostics";
+
+export class CosmosException extends Error {
+  response?: ErrorResponse;
+
+  constructor(m?: any, errorResponse?: ErrorResponse) {
+    super(m);
+    Object.setPrototypeOf(this, CosmosException.prototype);
+    this.response = errorResponse;
+    if (typeof m !== "undefined") {
+      setDiagnostics(m);
+    }
+  }
+  getDiagnostics() {
```

Comment:
shouldn't this also return duarion, regions contacted etc.? Sp Error + ComsosDiagnostic

Reply:
done